pkg/diff/diff.go

```diff
@@ -506,30 +506,36 @@ func compareData(map1, map2 map[string][]byte, null1, null2 map[string]bool, ord
 			return false, 0, errors.Errorf("don't have key %s", col.Name.O)
 		}
 		if needQuotes(col.FieldType) {
-			if string(data1) > string(data2) {
-				cmp = 1
-				break
-			} else if string(data1) < string(data2) {
-				cmp = -1
-				break
-			} else {
+
+			strData1 := string(data1)
+			strData2 := string(data2)
+
+			if strData1 == strData2 {
```

Comment:
I think it is better to compare string length first: `if len(strData1) == len(strData2) && ...`

Reply:
done